asyncpg/connect_utils.py

```diff
@@ -601,6 +601,11 @@ def connection_lost(self, exc):
         raise asyncio.TimeoutError
 
     connected = _create_future(loop)
+
+    params_input = params
+    if callable(params.password):
```

Comment:
Please also add support for coroutines (via `inspect.iscoroutinefunction()`) to support async callbacks.

Reply:
Good point, updated the branch with support for async password callbacks and an additional test for that case as well.